### PR TITLE
feat: add copy function to visibility input

### PR DESCRIPTION
**Pull request by anh-ld**
I added a copy button to the visibility input so people can quickly copy text value to clipboard in just 1 click. I mainly use this for quickly get the database's external host url.

### CODE REVIEW

File: components/shared/toggle-visibility-input.tsx [hunk 1]
```diff
@@ -1,26 +1,44 @@
...
-		</div>
-	);
+  const copyToClipboard = () => {
+    if (!inputRef.current) return;
+
+    const inputElement = inputRef.current;
+    const text = inputElement.value;
+
+    inputElement.select();
+    inputElement.setSelectionRange(0, 99999);
```
Comment (kdurek) [inline components/shared/toggle-visibility-input.tsx]: is it safe?

Comment (anh-ld) [reply components/shared/toggle-visibility-input.tsx]: It is. Just a function to copy the input text value into clipboard. [Here](https://www.w3schools.com/howto/howto_js_copy_clipboard.asp) is the reference.

Comment (kdurek) [reply components/shared/toggle-visibility-input.tsx]: Yeah i just wondered for the 9999 number? Shouldn't select alone be enough?

Comment (kdurek) [reply components/shared/toggle-visibility-input.tsx]: https://developer.mozilla.org/en-US/docs/Web/API/HTMLInputElement/select

In your link they say its for mobile, in this link it Works perfectly fine with only select and on mobile 😁

**Comment by kdurek** [comment]
I got another one, maybe can we get it generally available on input with passing prop 'withCopy' or something so we can use of elsewhwre

File: components/shared/toggle-visibility-input.tsx (hunk 1 above)
Comment (anh-ld) [reply components/shared/toggle-visibility-input.tsx]: Oh, you're right. I think the `setSelectionRange` is needed when the `select` is not supported, which happened quite a long time ago.

Now, using the `setSelectionRange` is redundant. I have updated my PR to remove it.

**Comment by anh-ld** [comment]
> I got another one, maybe can we get it generally available on input with passing prop 'withCopy' or something so we can use of elsewhwre

I don't think it's needed atm. The `input` component now is as simple as it should be. IMO, `input with copy` should be another component else that serve a certain purpose.

**Comment by kdurek** [comment]
> > I got another one, maybe can we get it generally available on input with passing prop 'withCopy' or something so we can use of elsewhwre
> 
> 
> 
> I don't think it's needed atm. The `input` component now is as simple as it should be. IMO, `input with copy` should be another component else that serve a certain purpose.

Ok good enough 😁

File: components/shared/toggle-visibility-input.tsx [hunk 2]
```diff
@@ -1,19 +1,36 @@
...
+
+		const inputElement = inputRef.current;
+		const text = inputElement.value;
+
+		inputElement.select();
+		navigator.clipboard.writeText(text);
+
+		toast.success("Value is copied to clipboard");
+	};
+
```
Comment (Siumauricio) [inline components/shared/toggle-visibility-input.tsx]: We use a `copy-to-clipboard` package, we can remove this function and use instead:

`import copy from "copy-to-clipboard";`

Then 
```ts
copy("String");
```

**Review by Siumauricio — CHANGES_REQUESTED** [review]

File: components/shared/toggle-visibility-input.tsx (hunk 2 above)
Comment (anh-ld) [reply components/shared/toggle-visibility-input.tsx]: Yes. That'd be good. I have updated the PR.

**Review by Siumauricio — APPROVED** [review]
Thank you ❤️

**Pull request merged**